src/resource.js

```diff
@@ -1,6 +1,58 @@
 'use strict'
 
-export default () => {
+import url from 'url'
+import jts from 'jsontableschema'
+import _ from 'lodash'
 
-}
+export default class Resource {
 
+  constructor(descriptor) {
+    this._descriptor = descriptor
+    this.REMOTE_PROTOCOLS = ['http:', 'https:', 'ftp:', 'ftps:']
```

Comment:
That is my thought as well, but I settled for including it here for now and make something similar to  [utilities from jts](https://github.com/frictionlessdata/jsontableschema-js/blob/master/src/utilities.js as) as I make progress.

Maybe other possibility is to export the `utulities` from jts and use them here?